### ISSUE TITLE
Archiver fails to write temporary files in rootless docker image

### ISSUE THREAD

**Issue by PKizzle** [go-gitea/gitea#15921]
- Gitea version (or commit ref): 1.14.2
- Git version: 2.30.2
- Operating system: Debian GNU/Linux 10 (buster)
  
  
  
- Database (use `[x]`):
  - [x] PostgreSQL
  - [ ] MySQL
  - [ ] MSSQL
  - [ ] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [ ] Yes (provide example URL)
  - [x] No (unable to view logs)
- Log gist:

```
2021/05/17 11:31:28 ...archiver/archiver.go:220:doArchive() [E] Unable to create a temporary archive file! Error: open /tmp/archive281431414: read-only file system
2021/05/17 11:31:31 ...archiver/archiver.go:220:doArchive() [E] Unable to create a temporary archive file! Error: open /tmp/archive850285149: read-only file system
2021/05/17 11:31:34 ...archiver/archiver.go:220:doArchive() [E] Unable to create a temporary archive file! Error: open /tmp/archive702526488: read-only file system
2021/05/17 11:31:37 ...archiver/archiver.go:220:doArchive() [E] Unable to create a temporary archive file! Error: open /tmp/archive350529687: read-only file system
2021/05/17 11:31:40 ...archiver/archiver.go:220:doArchive() [E] Unable to create a temporary archive file! Error: open /tmp/archive723930890: read-only file system
2021/05/17 11:31:43 ...archiver/archiver.go:220:doArchive() [E] Unable to create a temporary archive file! Error: open /tmp/archive967582945: read-only file system
```

## Description

When using the new rootless docker image gitea is unable to write into the `/tmp` directory since that requires root access.
Instead the archiver should use the specified `/tmp/gitea` folder to write temporary data to.

I think it is this line that needs to be changed: https://github.com/go-gitea/gitea/blob/ab77a24f18ae45248bd6c71778fe3bfacacf864f/services/archiver/archiver.go#L218

**Comment by lunny** [comment]
Since v1.15 have rework the archiver, it's a problem of v1.14 special.

**Comment by PKizzle** [comment]
Is it possible to backport the changes to v1.14?

**Comment by lunny** [comment]
No. I don't think so. But we can send a PR to v1.14 to fix that based on v1.14 code.

**Comment by wxiaoguang** [comment]
Outdated
